### PR TITLE
ISPN-5231 CacheNotifierImpl should wire the listener and converter with ...

### PR DESCRIPTION
**Pull request by anistor**
...dependecies from ComponentRegistry

Jira: https://issues.jboss.org/browse/ISPN-5231
        https://issues.jboss.org/browse/ISPN-5257
        https://issues.jboss.org/browse/ISPN-5300

### CODE REVIEW

File: core/src/main/java/org/infinispan/notifications/cachelistener/CacheNotifierImpl.java [hunk 1]
```diff
@@ -652,7 +651,7 @@ public void addListener(Object listener, KeyFilter<? super K> filter, ClassLoade
       builder
             .setIncludeCurrentState(l.includeCurrentState())
             .setClustered(l.clustered())
-            .setOnlyPrimary(l.clustered() ? (cacheMode.isDistributed() ? true : false) : l.primaryOnly())
+            .setOnlyPrimary(l.clustered() ? cacheMode.isDistributed() : l.primaryOnly())
```
Comment (wburns) [inline core/src/main/java/org/infinispan/notifications/cachelistener/CacheNotifierImpl.java]: Was this change an oversight or was there some issue? This was specifically for REPL caches so it only notifies on the write to the node where the listener is notified.

Comment (anistor) [inline core/src/main/java/org/infinispan/notifications/cachelistener/CacheNotifierImpl.java]: I simplified ```(cacheMode.isDistributed() ? true : false)``` :)

Comment (wburns) [inline core/src/main/java/org/infinispan/notifications/cachelistener/CacheNotifierImpl.java]: Ah yes sorry, misread :)

**Comment by anistor** [comment]
I've just found a minor improvement and updated this.

**Comment by anistor** [comment]
Hold this please! Just noticed the problem is only half solved. It still does not work if ```includeCurrentState = true```. I'll let you know when it's fixed.

**Comment by anistor** [comment]
@wburns Should be ready now.

File: core/src/main/java/org/infinispan/filter/CompositeKeyFilter.java [hunk 2]
```diff
@@ -19,4 +30,43 @@ public boolean accept(K key) {
...
+   public static class Externalizer extends AbstractExternalizer<CompositeKeyFilter> {
+
+      @Override
+      public Set<Class<? extends CompositeKeyFilter>> getTypeClasses() {
+         return Util.<Class<? extends CompositeKeyFilter>>asSet(CompositeKeyFilter.class);
+      }
+
+      @Override
+      public void writeObject(ObjectOutput output, CompositeKeyFilter object) throws IOException {
+         output.writeInt(object.filters.length);
```
Comment (wburns) [inline core/src/main/java/org/infinispan/filter/CompositeKeyFilter.java]: We should be able to use the writeUnsignedInt method from UnsignedNumeric class.

File: core/src/main/java/org/infinispan/filter/CompositeKeyValueFilter.java [hunk 3]
```diff
@@ -34,6 +35,13 @@ public boolean accept(K key, V value, Metadata metadata) {
...
    }
 
+   @Inject
+   protected void injectDependencies(ComponentRegistry cr) {
+      for (KeyValueFilter<? super K, ? super V> f : filters) {
+         cr.wireDependencies(f);
+      }
+   }
+
    public static class Externalizer extends AbstractExternalizer<CompositeKeyValueFilter> {
```
Comment (wburns) [inline core/src/main/java/org/infinispan/filter/CompositeKeyValueFilter.java]: Technically this Externalizer can use the UnsignedNumeric as well, sorry about missing this originally.

**Comment by wburns** [comment]
It looks fine other than the minor suggestion.

**Comment by wburns** [comment]
Pulling..

**Comment by wburns** [comment]
Integrated into master, thanks @anistor !

**Pull request closed without merging**